server/app/services/applicant/question/FileUploadQuestion.java

```diff
@@ -40,6 +44,23 @@ public ImmutableList<Path> getAllPaths() {
     return ImmutableList.of();
   }
 
+  @Override
+  public ImmutableMap<Path, String> getJsonEntries() {
+    return ImmutableMap.of(
+        applicantQuestion.getContextualizedPath().join(Scalar.FILE_KEY),
+        applicantQuestion
+            .createFileUploadQuestion()
+            .getFileKeyValue()
+            .map(
+                fileKey ->
+                    baseUrl
+                        + controllers.routes.FileController.adminShow(
+                                applicantQuestion.getProgramDefinitionId(),
+                                URLEncoder.encode(fileKey, StandardCharsets.UTF_8))
```

Comment:
**suggestion (non-blocking):** Would be preferable to use [FileController.acledAdminShow](https://github.com/civiform/civiform/blob/main/server/app/controllers/FileController.java#L88) (see comments in that controller)

Reply:
Done.